common/test/acceptance/tests/lms/test_problem_types.py

```diff
@@ -0,0 +1,752 @@
+        And a "problem_check" browser event is emitted
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Answer the problem correctly
+        self.answer_problem(True)
+        self.problem_page.click_check()
+        self.wait_for_status('correct')
+
+        # Check for corresponding tracking event
+        expected_events = [
+            {'event_source': 'server', 'event_type': 'problem_check'},
+            {'event_source': 'browser', 'event_type': 'problem_check'},
+        ]
+
+        for event in expected_events:
+            self.wait_for_events(event_filter=event, number_of_matches=1)
+
+    @attr('shard_2')
+    def test_answer_incorrectly(self):
+        """
+        Scenario: I can answer a problem incorrectly
+        Given External graders respond "incorrect"
+        And I am viewing a "<ProblemType>" problem
+        When I answer a "<ProblemType>" problem "incorrectly"
+        Then my "<ProblemType>" answer is marked "incorrect"
+        And The "<ProblemType>" problem displays a "incorrect" answer
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Answer the problem incorrectly
+        self.answer_problem(False)
+        self.problem_page.click_check()
+        self.wait_for_status('incorrect')
+
+    @attr('shard_2')
+    def test_submit_blank_answer(self):
+        """
+        Scenario: I can submit a blank answer
+        Given I am viewing a "<ProblemType>" problem
+        When I check a problem
+        Then my "<ProblemType>" answer is marked "incorrect"
+        And The "<ProblemType>" problem displays a "blank" answer
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Leave the problem unchanged and click check.
+        self.problem_page.click_check()
+        self.wait_for_status('incorrect')
+
+    @attr('a11y')
+    def test_problem_type_a11y(self):
+        """
+        Run accessibility audit for the problem type.
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Set the scope to the problem container
+        self.problem_page.a11y_audit.config.set_scope(
+            include=['div#seq_content'])
+
+        # Run the accessibility audit.
+        self.problem_page.a11y_audit.check_for_accessibility_errors()
+
+
+class CheckboxProblemTypeTest(ProblemTypeTestBase, ProblemTypeTestMixin):
+    """
+    TestCase Class for Checkbox Problem Type
+    """
+    problem_name = 'CHECKBOX TEST PROBLEM'
+    problem_type = 'checkbox'
+
+    factory = ChoiceResponseXMLFactory()
+
+    factory_kwargs = {
+        'question_text': 'The correct answer is Choice 3',
```

Comment:
Reading the `answer_problem` it looks like the correct answer is actually both Choices 1 and 3?